### PR TITLE
Add another docstring formatter in pre-commit

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

@Pierre-Sassoulas Just to show off the new program I wrote. It is somewhat nitpicky, but would save time in review. Repository can be found [here](https://github.com/DanielNoord/pydocstringformatter).
Let me know if this is something you would want to merge or if anything is missing!

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -95,3 +95,8 @@ repos:
       - id: prettier
         args: [--prose-wrap=always, --print-width=88]
         exclude: tests(/.*)*/data
+  - repo: https://github.com/DanielNoord/pydocstringformatter
+    rev: v0.1.0
+    hooks:
+      - id: pydocstringformatter
+        exclude: tests/functional/|tests/input|tests(/.*)*/data|tests/regrtest_data/|tests/data/
```
Comment (Pierre-Sassoulas) [inline .pre-commit-config.yaml:102]:
```suggestion
        exclude: *fixtures
```
Or is that different than the fixture we have otherwise ?

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
Looks pretty good ! Would you mind checking the updated #5624 ? I removed docformatter and applied pydocstringformatter. I've done some (hundreds 😄) manual fixes here, I'd like them to be added if possible. But I think there's some case where pydocstringformatter do not fix some issue see https://github.com/PyCQA/pylint/pull/5624/files#diff-35d3c5349e5876e9820da739cbefc979ef433a55ab338df84cae9e2cd1b0d0c2R6 for example.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (DanielNoord) [reply .pre-commit-config.yaml]: I didn't know about this!

**Comment by DanielNoord** [comment]
I fixed the issue with module docstrings not getting recognised.

I have got some other things I would like to put in `0.2.0` so I haven't released it yet, but that should be good to go somewhere the coming days. I'll use this PR to update to `0.2.0`. After that I can look at #5624. If my program works the only thing that should remain in that PR are your manual fixes 😄

**Comment by Pierre-Sassoulas** [comment]
To be noted that when we add it's going to be independent, there will be a an automatic upgrade if you release a version of ``pydocstringformatter``  by the pre-commit bot without us doing anything about it. It's still a good idea to wait a little so we don't have too much formatting related changes when upgrading. 

Also maybe a cool name would be nice ? 😄 It's similar to black so something with color ? It's a shame that 257 is too high for RGB and PEP too high for hex code, but #FEF257 is close enough and is a light yellow, so maybe ``lemon``, ``custard``, ``golden`` or ``chanteclerc`` 😉

fef257

**Comment by DanielNoord** [comment]
> To be noted that when we add it's going to be independent, there will be a an automatic upgrade if you release a version of `pydocstringformatter` by the pre-commit bot without us doing anything about it. It's still a good idea to wait a little so we don't have too much formatting related changes when upgrading.

Indeed! I think the current `dev` version is good enough for us, but I'd like to add some better options registering for potential other users.

> Also maybe a cool name would be nice ? 😄 It's similar to black so something with color ? It's a shame that 257 is too high for RGB and PEP too high for hex code, but #FEF257 is close enough and is a light yellow, so maybe `lemon`, `custard`, `golden` or `chanteclerc` 😉
> 
> fef257

I'm open to good suggestions, I think `chanteclerc` is quite good 😄

**Comment by Pierre-Sassoulas** [comment]
(Chanteclerc is a variety of yellow apples.) Though I realized that basically every pep would be very close in color to each other with this 'pep to hex' reasoning.

Another possibility is checking what happened in 257 (a lot had to do with Valerian, which sounds Nice) or 257 bc (Tyndaris, sounds nice too). https://en.m.wikipedia.org/wiki/257_BC.

**Comment by DanielNoord** [comment]
@Pierre-Sassoulas Updated to `0.2.0`. That should fix the issue you identified and also format single line docstrings a little better. I force pushed to create a sane commit history, we should now be able to rebase and merge this.

**Review by Pierre-Sassoulas — APPROVED** [review]
That's pretty nice 👍

**Pull request merged**